File: frontend/src/services/announcement.service.js
import { announcementApi } from '../api/announcement.api';

export const announcementService = {
  async list() {
    const res = await announcementApi.getAll();
    return res.data || [];
  },
  async create(payload) {
    const res = await announcementApi.create(payload);
    return res.data;
  },
  async deactivate(id) {
    await announcementApi.deactivate(id);
  },
};

File: src/main/java/com/hrms/hr_payroll_management_system/service/AnnouncementService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.dto.request.announcement.CreateAnnouncementRequest;
import com.hrms.hr_payroll_management_system.dto.response.announcement.AnnouncementResponse;

import java.util.List;

public interface AnnouncementService {

    AnnouncementResponse create(CreateAnnouncementRequest request, String postedByEmail);

    List<AnnouncementResponse> getRecent(int limit);

    List<AnnouncementResponse> getAll();

    void deactivate(Long id);
}